File: C++/1291. Sequential Digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high)
    {
      vector<int> result;

        for (int i = 1; i <= 9; ++i) {
            int current = i;
            while (current <= high && current % 10 != 0) {
                if (current >= low) {
                    result.push_back(current);
                }
                current = current * 10 + (current % 10 + 1);
            }
        }

        std::sort(result.begin(), result.end());
        return result;  
    }
};

File: JAVA/1291. Sequential Digits.java
class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        String str="123456789";
        List<Integer> res=new ArrayList<>();
        for(int i=0;i<str.length();i++){
            for(int j=i+1;j<=str.length();j++){
                String substr=str.substring(i,j);
                int num=Integer.parseInt(substr);
                if(num>=low && num<=high){
                    res.add(num);
                }
            }
        }
        Collections.sort(res);
        return res;
    }
}
